src/status_im/contexts/wallet/data_store_test.cljs

```diff
@@ -0,0 +1,193 @@
+                                    :path                     (:path raw-account)
+                                    :name                     (:name raw-account)
+                                    :wallet                   (:wallet raw-account)
+                                    :removed                  (:removed raw-account)
+                                    :color                    (keyword (:colorId raw-account))
+                                    :created-at               (:createdAt raw-account)
+                                    :prod-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:prodPreferredChainIds raw-account))
+                                    :test-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:testPreferredChainIds
+                                                                raw-account))}]}
+             (sut/rpc->keypair raw-keypair--private-key)))))
+
+(deftest rpc->keypairs-test
+  (testing "rpc->keypairs transforms and sorts raw keypairs"
+    (is
+     (match? [(sut/rpc->keypair raw-keypair--profile)
+              (sut/rpc->keypair raw-keypair--seed-phrase)
+              (sut/rpc->keypair raw-keypair--private-key)]
+             (sut/rpc->keypairs [raw-keypair--seed-phrase
+                                 raw-keypair--private-key
+                                 raw-keypair--profile])))))
+
+(deftest process-keypairs-test
+  (testing "process-keypairs represents updated key pairs and accounts"
+    (is
+     (match?
+      (matchers/match-with
+       [set? matchers/set-equals
+        map? matchers/equals]
+       {:removed-keypair-ids         #{}
+        :removed-account-addresses   #{}
+        :updated-accounts-by-address (->>
+                                       [raw-keypair--seed-phrase
+                                        raw-keypair--private-key]
+                                       (sut/rpc->keypairs)
+                                       (mapcat :accounts)
+                                       (utils.collection/index-by :address))
+        :updated-keypairs-by-id      {(:key-uid raw-keypair--seed-phrase)
+                                      (sut/rpc->keypair raw-keypair--seed-phrase)
+                                      (:key-uid raw-keypair--private-key)
+                                      (sut/rpc->keypair raw-keypair--private-key)}})
+      (sut/process-keypairs [raw-keypair--seed-phrase
+                             raw-keypair--private-key]))))
+  (testing "process-keypairs represents removed key pairs and accounts"
+    (is (match?
+         (matchers/match-with
+          [set? matchers/set-equals
+           map? matchers/equals]
+          {:removed-keypair-ids         #{"0x456"}
+           :removed-account-addresses   #{"1x456"}
+           :updated-accounts-by-address (->> [raw-keypair--seed-phrase]
```

Comment:
This is sort of an anti-pattern in unit tests, otherwise the test would need tests for itself. It's also harder for the reader to understand the test output without actually running the test. I know it comes from the desire to keep tests more DRY, but we can also eliminate duplication from the final output directly, thus ignoring intermediary steps.

Reply:
Hey @ilmotta thanks for mentioning this, I was thinking something similar when writing this code, especially since the tests are nearly re-implementing the logic. I'm okay with running this expression inside the REPL and replacing the expression with data.
```
(->> [raw-keypair--seed-phrase]
     (sut/rpc->keypairs)
     (mapcat :accounts)
     (utils.collection/index-by :address))
```

Reply:
As a side note, do we know if Malli would help here with generating shapes of data like this?

Reply:
Malli would definitely be able to help @seanstrom. I experimented a lot with generators in our unit tests before Malli was integrated in `status-mobile`. You would then basically define a helper function in your test namespace that generates whatever you need, grabbing the schema from malli's registry or from a var depending on where we would define the schemas, and then this utility function would assoc or do a nested merge (I remember I had to add this utility) to conveniently allow each test to replace parts of the generated data.

One important bit is that I was spec'ing parts of the app-db in my experiments so that I didn't need to write schemas only for testing, which seems to be a point of tension in our teams 🤷🏼‍♂️